### ISSUE TITLE
pretty high lenght of time since installation or upgrade

### ISSUE THREAD

**Issue by Productivix** [react-native-device-info/react-native-device-info#1431]
## Summary
|             |     |
| RN |0.68.2 |
| Version     | 9.0.9   |
| Affected OS | Android   |
| OS Version  | any from 8 to 11  |

## Current behavior

I calculate the time of my app since the last installation or upgrade (same strange behavior)
The time is pretty high : I get 19164 days for an app that is tested since the 07th of June of this year .

## Expected behavior
I would expect a time of 23-7 = 16 days 

### The Code : 

``` 
useEffect(() => {
    //depuis combien de secondes est-ce installé ?
    const depuisInstall = async () => {
      try {
        let secinstall = await DeviceInfo.getLastUpdateTime();
        let jours = Math.floor(secinstall / (3600000 * 24));
        console.log('en jours:', jours); // il y a un pb !
        return jours; //le retour ne marche pas ..{"_U": 0, "_V": 0, "_W": null, "_X": null}
      } catch (error) {
        console.log(error.message);
      }
    };
    depuisInstall();
  }, []);
```

Maybe I misunderstood the command ?
thanks  a lot for this lib !

**Comment by mikehardy** [comment]
:thinking: 

Perhaps there is some math issue (overflow or something) or device clock issue (device times are not under your control...) but the code we use to populate the information is direct from the system and quite basic:

https://github.com/react-native-device-info/react-native-device-info/blob/0341ab5b5909a410d40d168832f415b93bd557be/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java#L666-L686

On an emulator using the sample app installed once then re-installed a bit more than a minute later the difference is as expected

![image](https://user-images.githubusercontent.com/782704/175316893-18b27321-b97a-40b3-8c62-dda0b716d8bd.png)


So, cannot reproduce and will close. If you can reproduce this with the sample app and direct math on the numbers from a device where you know the clock is working correctly on first install and updates we can reopen.

**Comment by Productivix** [comment]
hi, I tested your example as described on : https://github.com/react-native-device-info/react-native-device-info/tree/master/example
but may be my installation is too recent , 
I get : 
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :app:processDebugMainManifest FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.2/userguide/command_line_interface.html#sec:command_line_warnings
20 actionable tasks: 20 executed
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugMainManifest'.
> Unable to make field private final java.lang.String java.io.File.path accessible: module java.base does not "opens java.io" to unnamed module @59d3a156

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2m 20s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01

FAILURE: Build failed with an exception.

So I cannot compare and tell you !

**Comment by Productivix** [comment]
By the way I tested the issue on 2 different terminals and Android versions , with same problem.

**Comment by Productivix** [comment]
thank you @mikehardy ! That is why I add some projects with native calls and similar messages - it is written " make sure it is version 11 or newer." on https://reactnative.dev/docs/0.68/environment-setup , I took newer ! - I will decrease to JDK11 so !

yes for my issue , I mixed time dans duration : here is an example of application : 
'''
 useEffect(() => {
    //for how many minutes is it installed ? and with a maxi of 30 days try, how many remains ?
    
const mnInstall = async () => {
      try {
        let secinstall = await DeviceInfo.getFirstInstallTime();
        setSecOfLastInstall(secinstall);
        let delaiMn = Math.floor((Date.now() - secinstall) / 60000);
        console.log("delai en mn de l'install:", delaiMn);
        setNbrJoursRestantTest(
          Math.floor((30 * 24 * 60 - delaiMn) / (24 * 60)),
        );
      } catch (error) {
        console.log(error.message);
      }
    };
    mnInstall();
    
  }, []);
'''

**Comment by mikehardy** [comment]
@Productivix I proposed a change to the developer docs that restricts the range of JDKs from 11 to 17 inclusive instead of the open-ended "...or newer" since JDK18 does not work. Hopefully that will help others avoid the trap. Good luck with your project!

**Comment by Productivix** [comment]
very good and best  !- yes all but some projets were working in JDK17

**Comment by mikehardy** [comment]
@Productivix can you be more specific about this?

> all but some projets were working in JDK17

I am not sure I understand so I am just looking for clarity. I could read this as:

1) "all of our projects were on JDK17 except some (which were on JDK18) and they work"

or maybe

2) "some of our projects work with JDK17 but some projects do not work with JDK17"

That's important for the documentation change because I proposed JDK11 through JDK17 right now, but perhaps there are problems with JDK17 in some projects too...

**Comment by Productivix** [comment]
HI Mike, 
to be more precise about that JDK, :
3 days ago, I reported you gradle errors on your example test with JDK17.
I had the same error with "old" projects cloned since 13 April 22 (I installed JDK17). But made >15 smalll projects in RN 68.2 without any errors.

Today I remake with JDK11 the same test from scratch https://github.com/react-native-device-info/react-native-device-info/tree/master/example
and only get the error  : (to be corrected I think, but not the same level)
`
 BUNDLE  ./index.js

 LOG  Running "example" with {"rootTag":1}
 ERROR  TypeError: _reactNativeDeviceInfo.default.getBrightnessSync is not a function. (In '_reactNativeDeviceInfo.default.getBrightnessSync()', '_reactNativeDeviceInfo.default.getBrightnessSync' is undefined)

This error is located at:
    in App (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in example(RootComponent) (at renderApplication.js:60)
 ERROR  TypeError: _reactNativeDeviceInfo.default.getBrightnessSync is not a function. (In '_reactNativeDeviceInfo.default.getBrightnessSync()', '_reactNativeDeviceInfo.default.getBrightnessSync' is undefined)

This error is located at:
    in App (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in example(RootComponent) (at renderApplication.js:60)
`
So I think that for new projects, JDK 17 is ok , but to test projects built with RN<=.67, JDK11 is better.
So the correction in the documentation has to be made to avoid open-ended .
Hope it helps.
